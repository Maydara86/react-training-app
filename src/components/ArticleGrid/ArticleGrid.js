import React from 'react'
import styles from './ArticleGrid.module.css'

export default function ArticleGrid(props) {
  return <div className={styles.container}>{props.children}</div>
}
