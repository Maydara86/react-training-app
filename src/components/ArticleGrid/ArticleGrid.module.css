.container {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr;
    grid-auto-columns: 1fr 1fr 1fr;
    grid-gap: 1em 1em;
    place-items: center center;
    place-content: center center;
}