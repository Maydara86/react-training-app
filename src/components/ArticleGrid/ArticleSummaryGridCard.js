import React from 'react'
import User from '../User/User'
import myImage from '../../assets/images/article.jpeg'
import { ClapFilled, ClapUnfilled, BookmarkFilled, BookmarkUnfilled } from '../Svg/Svg'
import styles from './ArticleSummaryGridCard.module.css'

export default function ArticleSummaryGridCard({
  id,
  articleName,
  claps,
  bookmark,
  user,
  didClap,
  handleClapClick,
  handleBookmarkClick,
  handleStarClick,
  handleFollowClick,
}) {
  const gridTheme = {
    '--article-width': '400px',
    '--article-height': '440px',
    '--article-title-color': 'rgba(0, 0, 0, 0.84)',
    '--article-title-fontweight': '500',
    '--article-border': 'none',
  }

  // eslint-disable-next-line array-callback-return
  Object.keys(gridTheme).map(key => {
    const value = gridTheme[key]
    document.documentElement.style.setProperty(key, value)
  })

  const bookmarkFilled = (
    <BookmarkFilled
      className={styles.bookmarkIconItem}
      key={id}
      bookmarkClick={handleBookmarkClick}
      id={id}
    />
  )

  const bookmarkUnfilled = (
    <BookmarkUnfilled
      className={styles.bookmarkIconItem}
      key={id}
      bookmarkClick={handleBookmarkClick}
      id={id}
    />
  )

  const clapFilled = (
    <ClapFilled className={styles.clapsIconItem} handleClapClick={handleClapClick} id={id} />
  )

  const clapUnfilled = (
    <ClapUnfilled className={styles.clapsIconItem} handleClapClick={handleClapClick} id={id} />
  )

  return (
    <div key={id} className={styles.articleContainer}>
      <img src={myImage} alt={articleName} className={styles.articleImageItem} />
      <h3 className={styles.articleTitleItem}>{articleName}</h3>
      <div className={styles.authorAndLikesItem}>
        <div className={styles.authorAndLikesContainer}>
          <User
            className={styles.authorItem}
            {...user}
            handleFollowClick={handleFollowClick}
            handleStarClick={handleStarClick}
          />
          <div className={`${styles.likesItem} ${styles.likesContainer}`}>
            {didClap ? clapFilled : clapUnfilled}
            <div className={styles.clapsNumberItem}>{claps}</div>
            <span className={styles.separatorItem} />
            {bookmark ? bookmarkFilled : bookmarkUnfilled}
          </div>
        </div>
      </div>
    </div>
  )
}
