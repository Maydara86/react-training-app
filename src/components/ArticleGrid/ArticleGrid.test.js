import React from 'react'
import renderer from 'react-test-renderer'
import toJson from 'enzyme-to-json'
import { mount, shallow, render } from '../../enzyme'
import ArticleGrid from './ArticleGrid'

jest.mock('../Article/Article', () => 'Article')
jest.mock('../User/User', () => 'User')
jest.mock('../Svg/Svg', () => 'ClapFilled')
jest.mock('../Svg/Svg', () => 'ClapUnfilled')
jest.mock('../Svg/Svg', () => 'BookmarkFilled')
jest.mock('../Svg/Svg', () => 'BookmarkUnfilled')
jest.mock('../Svg/Svg', () => 'Star')

describe('ArticleGrid', () => {
  xit('renders corrctly using mount', () => {
    const wrapper = mount(<ArticleGrid />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  xit('renders corrctly using shallow', () => {
    const articleGrid = shallow(<ArticleGrid />)
    expect(articleGrid).toMatchSnapshot()
  })

  xit('renders corrctly using enzym render', () => {
    const articleGridRender = render.create(<ArticleGrid />)
    expect(articleGridRender).toMatchSnapshot()
  })

  xit('renders correctly using jest renderer', () => {
    const tree = renderer.create(<ArticleGrid />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
