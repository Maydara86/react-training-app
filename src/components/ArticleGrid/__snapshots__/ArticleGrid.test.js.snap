// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleGrid renders corrctly using mount 1`] = `"<div class=\\"container\\"></div>"`;

exports[`ArticleGrid renders corrctly using mount 2`] = `ShallowWrapper {}`;

exports[`ArticleGrid renders correctly using jest renderer 1`] = `
<div
  className="container"
/>
`;
